Libraries/LibJS/AST.cpp

```diff
@@ -658,6 +665,100 @@ Value UnaryExpression::execute(Interpreter& interpreter, GlobalObject& global_ob
     ASSERT_NOT_REACHED();
 }
 
+Value SuperExpression::execute(Interpreter&, GlobalObject&) const
+{
+    // The semantics for SuperExpressions are handled in CallExpression::compute_this_and_callee()
+    ASSERT_NOT_REACHED();
+}
+
+Value ClassMethod::execute(Interpreter& interpreter, GlobalObject& global_object) const
+{
+    return m_function->execute(interpreter, global_object);
+}
+
+Value ClassExpression::execute(Interpreter& interpreter, GlobalObject& global_object) const
+{
+    Value class_constructor_value = m_constructor->execute(interpreter, global_object);
+    if (interpreter.exception())
+        return js_undefined();
+    update_function_name(class_constructor_value, m_name);
+
+    ASSERT(class_constructor_value.is_function() && class_constructor_value.as_function().is_script_function());
+    ScriptFunction* class_constructor = static_cast<ScriptFunction*>(&class_constructor_value.as_function());
+    Value super_constructor = js_undefined();
+    if (!m_super_class.is_null()) {
+        super_constructor = m_super_class->execute(interpreter, global_object);
+        if (interpreter.exception())
+            return {};
+        if (!super_constructor.is_function() && !super_constructor.is_null())
+            return interpreter.throw_exception<TypeError>(ErrorType::ClassDoesNotExtendAConstructorOrNull, super_constructor.to_string_without_side_effects().characters());
+
+        class_constructor->set_constructor_kind(Function::ConstructorKind::Derived);
+        Object* prototype = Object::create_empty(interpreter, interpreter.global_object());
+        prototype->set_prototype(super_constructor.is_null() ? nullptr : &super_constructor.as_object().get("prototype").as_object());
+        prototype->define_property("constructor", class_constructor, 0);
+        class_constructor->define_property("prototype", prototype, 0);
+        class_constructor->set_prototype(super_constructor.is_null() ? global_object.function_prototype() : &super_constructor.as_object());
+    }
+
+    auto class_prototype = class_constructor->get("prototype");
+    ASSERT(class_prototype.is_object());
```

Comment:
Same here, also we probably should handle non-object results more gracefully (i.e. an exception).

Reply:
I think it's a bug if the class constructor doesn't have a `prototype` property (i.e. we accidentally constructed a `ScriptFunction` with `is_arrow_function == true`).

Reply:
An exception doesn't necessarily mean it's missing, accessing the property on its own can throw - i.e. when intercepted using `Proxy`.

Reply:
Going to throw a `TypeError` just to be safe.